### PR TITLE
Search testing

### PR DESCRIPTION
**Pull request by rebekadittrich**

### CODE REVIEW

File: src/test/java/com/google/sps/data/EventStorageTest.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    entity.setProperty("invited-users", event.getInvitedIDs());
+    entity.setProperty("joined-users", event.getJoinedIDs());
+    entity.setProperty("declined-users", event.getDeclinedIDs());
+
+    return entity;
+  }
+
+  @Test
+  public void eventStorageTest_getSearchedEvents_nullSearch_MatchesAll() {
+    assertEquals(3, eventStorageObject.getSearchedEvents(NULL_SEARCH).size());
```
Comment (andru47) [inline src/test/java/com/google/sps/data/EventStorageTest.java]: Maybe you should verify if the getSearchedEvents returns what it needs to return

File: src/test/java/com/google/sps/data/EventStorageTest.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  @Test
+  public void eventStorageTest_getSearchedEvents_emptySearch_MatchesAll() {
+    assertEquals(3, eventStorageObject.getSearchedEvents(EMPTY_SEARCH).size());
```
Comment (andru47) [inline src/test/java/com/google/sps/data/EventStorageTest.java]: same as above

File: src/test/java/com/google/sps/data/EventStorageTest.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  @Test
+  public void eventStorageTest_getSearchedEvents_Matches() {
+    assertEquals(1, eventStorageObject.getSearchedEvents(SEARCH_A).size());
```
Comment (andru47) [inline src/test/java/com/google/sps/data/EventStorageTest.java]: Also this one and the next one

**Review by andru47 — APPROVED** [review]

**Review by LadyPython — APPROVED** [review]

File: src/test/java/com/google/sps/servlets/EventServletTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    when(mockRequest.getParameter("duration")).thenReturn(search.getDuration());
+    when(mockRequest.getParameter("location")).thenReturn(search.getLocation());
+  }
+
+  private void setWriterAndCreateEvent() throws IOException {
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+
+    new EventServlet(mockUserStorage, mockEventStorage).doGet(mockRequest, mockResponse);
```
Comment (marcbury-g) [inline src/test/java/com/google/sps/servlets/EventServletTest.java:105]: Don't hide this in the tests. It is counterintuitive but for tests it is more important that you can read them and understand them without going into each method call. Especially you should also check the output in your tests which you can't do now.

File: src/test/java/com/google/sps/servlets/EventServletTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    
+    setParameters(search);
+
+    setWriterAndCreateEvent();
+    
+    verify(mockEventStorage, atLeast(1)).getSearchedEvents(eq(search));
+  }
+
+  @Test
+  public void eventServletTest_doGet_WithSlashPathAndGivenParameters_DoesSearch() throws IOException {
```
Comment (marcbury-g) [inline src/test/java/com/google/sps/servlets/EventServletTest.java:180]: Please also add a test that the servlet writes the output correctly if something is returned in the `getSearchedEvents(...)` call.

**Review by marcbury-g — APPROVED** [review]
Just one minor thing.

**Pull request merged**